src/runtime/program/vote/execute.zig

```diff
@@ -138,10 +174,287 @@ fn intializeAccount(
         clock,
     );
     defer vote_state.deinit();
-    try vote_account.serializeIntoAccountData(vote_state);
+    try vote_account.serializeIntoAccountData(VoteStateVersions{ .current = vote_state });
+}
+
+/// Agave https://github.com/anza-xyz/agave/blob/0603d1cbc3ac6737df8c9e587c1b7a5c870e90f4/programs/vote/src/vote_processor.rs#L77-L79
+fn executeAuthorize(
+    allocator: std.mem.Allocator,
+    ic: *InstructionContext,
+    vote_account: *BorrowedAccount,
+    pubkey: Pubkey,
+    vote_authorize: VoteAuthorize,
+) InstructionError!void {
+    const clock = try ic.getSysvarWithAccountCheck(
+        Clock,
+        @intFromEnum(vote_instruction.Authorize.AccountIndex.clock_sysvar),
+    );
+
+    try authorize(
+        allocator,
+        ic,
+        vote_account,
+        pubkey,
+        vote_authorize,
+        clock,
+        null,
+    );
+}
+
+/// Agave https://github.com/anza-xyz/agave/blob/0603d1cbc3ac6737df8c9e587c1b7a5c870e90f4/programs/vote/src/vote_state/mod.rs#L678
+///
+/// Authorize the given pubkey to withdraw or sign votes. This may be called multiple times,
+/// but will implicitly withdraw authorization from the previously authorized
+/// key.
+fn authorize(
+    allocator: std.mem.Allocator,
+    ic: *InstructionContext,
+    vote_account: *BorrowedAccount,
+    authorized: Pubkey,
+    vote_authorize: VoteAuthorize,
+    clock: Clock,
+    signers: ?std.AutoHashMap(Pubkey, void),
+) InstructionError!void {
+    const versioned_state = try vote_account.deserializeFromAccountData(
+        allocator,
+        VoteStateVersions,
+    );
+    var vote_state = versioned_state.convertToCurrent(allocator) catch {
+        // TODO okay to convert out of memory to InvalidAccountData?
+        return InstructionError.InvalidAccountData;
+    };
+    defer vote_state.deinit();
+
+    switch (vote_authorize) {
+        .voter => {
+            const current_epoch = clock.epoch;
+            const target_epoch = std.math.add(u64, current_epoch, 1) catch {
+                return InstructionError.InvalidAccountData;
+            };
+
+            try checkAuthority(
```

Comment:
shouldnt this be `verifyAuthorizedSigner` ?

Reply:
Dropped in [694a8c6](https://github.com/Syndica/sig/pull/576/commits/694a8c64b1f56d8b9f4531069ea163b9d78be19d)

Reply:
The agave code has the same call. It is here https://github.com/anza-xyz/solana-sdk/blob/c07f692e41d757057c8700211a9300cdcd6d33b1/vote-interface/src/state/mod.rs#L872 

Only that it is not obvious, because it is called and passed to a closure that is passed as an argument to set_new_authorized_voter. See https://github.com/anza-xyz/agave/blob/131d3b3e007b1505231c726b98489aae40234a39/programs/vote/src/vote_state/mod.rs#L701

Reply:
ah i see okay thank you